web/src/components/storage/DASDFormatProgress.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) [2023] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React from "react";
+import { Progress, Stack } from "@patternfly/react-core";
+import { Popup } from "~/components/core";
+import { _ } from "~/i18n";
+import { useDASDDevices, useDASDRunningFormatJobs } from "~/queries/dasd";
+import { FormatSummary } from "~/types/dasd";
+
+export default function DASDFormatProgress() {
+  const devices = useDASDDevices();
+  const runningJobs = useDASDRunningFormatJobs().filter(
+    (job) => Object.keys(job.summary || {}).length > 0,
+  );
+
+  const ProgressContent = ({ progress }: { progress: { [key: string]: FormatSummary } }) =>
+    Object.entries(progress).map(([id, { total, step, done }]) => {
+      const device = devices.find((d) => d.id === id);
+
+      return (
+        <Progress
+          key={`progress_${id}`}
+          size="sm"
+          max={total}
+          value={step}
+          title={`${device.id} - ${device.deviceName}`}
+          measureLocation="none"
+          variant={done ? "success" : undefined}
+        />
+      );
+    });
+
+  return (
+    <Popup title={_("Formatting DASD devices")} isOpen={runningJobs.length > 0} disableFocusTrap>
+      {runningJobs.map((job) => {
+        return (
+          <Stack
+            key={`formatting_progress_${job.jobId}`}
+            hasGutter
+            className="dasd-format-progress"
+          >
+            <ProgressContent key={`progress_content_${job.jobId}`} progress={job.summary} />
+          </Stack>
+        );
+      })}
+    </Popup>
+  );
+}
```

Comment:
@teclator 

Sorry for the NP here. I've no way of testing below diff in which I have tried to improve just a bit this component by extracting the progress to an internal component outside of the component itself and ending up with just one `<Stack>`.

```diff
diff --git a/web/src/components/storage/DASDFormatProgress.tsx b/web/src/components/storage/DASDFormatProgress.tsx
index 61f2b4e6..bee2c2ac 100644
--- a/web/src/components/storage/DASDFormatProgress.tsx
+++ b/web/src/components/storage/DASDFormatProgress.tsx
@@ -24,7 +24,18 @@ import { Progress, Stack } from "@patternfly/react-core";
 import { Popup } from "~/components/core";
 import { _ } from "~/i18n";
 import { useDASDDevices, useDASDRunningFormatJobs } from "~/queries/dasd";
-import { FormatSummary } from "~/types/dasd";
+import { DASDDevice, FormatSummary } from "~/types/dasd";
+
+const DeviceProgress = ({ device, progress }: { device: DASDDevice, progress: FormatSummary }) => (
+  <Progress
+    size="sm"
+    max={progress.total}
+    value={progress.step}
+    title={`${device.id} - ${device.deviceName}`}
+    measureLocation="none"
+    variant={progress.done ? "success" : undefined}
+  />
+);
 
 export default function DASDFormatProgress() {
   const devices = useDASDDevices();
@@ -32,36 +43,19 @@ export default function DASDFormatProgress() {
     (job) => Object.keys(job.summary || {}).length > 0,
   );
 
-  const ProgressContent = ({ progress }: { progress: { [key: string]: FormatSummary } }) =>
-    Object.entries(progress).map(([id, { total, step, done }]) => {
-      const device = devices.find((d) => d.id === id);
-
-      return (
-        <Progress
-          key={`progress_${id}`}
-          size="sm"
-          max={total}
-          value={step}
-          title={`${device.id} - ${device.deviceName}`}
-          measureLocation="none"
-          variant={done ? "success" : undefined}
-        />
-      );
-    });
-
   return (
     <Popup title={_("Formatting DASD devices")} isOpen={runningJobs.length > 0} disableFocusTrap>
-      {runningJobs.map((job) => {
-        return (
-          <Stack
-            key={`formatting_progress_${job.jobId}`}
-            hasGutter
-            className="dasd-format-progress"
-          >
-            <ProgressContent key={`progress_content_${job.jobId}`} progress={job.summary} />
-          </Stack>
-        );
-      })}
+      <Stack
+        hasGutter
+        className="dasd-format-progress"
+      >
+        {runningJobs.flatMap((job) => (
+          Object.entries(job.summary).map(([id, progress]) => {
+            const device = devices.find((d) => d.id === id);
+            return <DeviceProgress key={`${id}-format-progress`} device={device} progress={progress} />;
+          }))
+        )}
+      </Stack>
     </Popup>
   );
 }
```

<details>
<summary>Click to show hide the whole file content</summary>

---

```tsx
/*
 * Copyright (c) [2023] SUSE LLC
 *
 * All Rights Reserved.
 *
 * This program is free software; you can redistribute it and/or modify it
 * under the terms of version 2 of the GNU General Public License as published
 * by the Free Software Foundation.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
 * more details.
 *
 * You should have received a copy of the GNU General Public License along
 * with this program; if not, contact SUSE LLC.
 *
 * To contact SUSE LLC about this file by physical or electronic mail, you may
 * find current contact information at www.suse.com.
 */

import React from "react";
import DASDTable from "~/components/storage/DASDTable";
import DASDFormatProgress from "~/components/storage/DASDFormatProgress";
import { _ } from "~/i18n";
import { Page } from "~/components/core";
import { useDASDDevicesChanges, useDASDFormatJobChanges } from "~/queries/dasd";

export default function DASDPage() {
  useDASDDevicesChanges();
  useDASDFormatJobChanges();

  return (
    <Page>
      <Page.Header>
        <h2>{_("DASD")}</h2>
      </Page.Header>

      <Page.MainContent>
        <DASDTable />
        <DASDFormatProgress />
      </Page.MainContent>
    </Page>
  );
}
```
</details>